nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/encryption/S3EncryptionService.java

```diff
@@ -0,0 +1,189 @@
+import org.apache.nifi.expression.ExpressionLanguageScope;
+import org.apache.nifi.processor.util.StandardValidators;
+
+import org.apache.nifi.processors.aws.s3.AbstractS3EncryptionService;
+import org.apache.nifi.processors.aws.s3.AbstractS3Processor;
+
+import org.apache.nifi.reporting.InitializationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+@Tags({"service", "encryption", "encrypt", "decryption", "decrypt", "key"})
+@CapabilityDescription("Adds configurable encryption to S3 Put and S3 Fetch operations.")
+public class S3EncryptionService extends AbstractControllerService implements AbstractS3EncryptionService {
+    private static final Logger logger = LoggerFactory.getLogger(S3EncryptionService.class);
+
+    public static final String STRATEGY_NAME_NONE = "NONE";
+    public static final String STRATEGY_NAME_SSE_S3 = "SSE_S3";
+    public static final String STRATEGY_NAME_SSE_KMS = "SSE_KMS";
+    public static final String STRATEGY_NAME_SSE_C = "SSE_C";
+    public static final String STRATEGY_NAME_CSE_KMS = "CSE_KMS";
+    public static final String STRATEGY_NAME_CSE_CMK = "CSE_CMK";
+
+    private static final Map<String, S3EncryptionStrategy> namedStrategies = new HashMap<String, S3EncryptionStrategy>() {{
+        put(STRATEGY_NAME_NONE, new NoOpEncryptionStrategy());
+        put(STRATEGY_NAME_SSE_S3, new ServerSideS3EncryptionStrategy());
+        put(STRATEGY_NAME_SSE_KMS, new ServerSideKMSEncryptionStrategy());
+        put(STRATEGY_NAME_SSE_C, new ServerSideCEKEncryptionStrategy());
+        put(STRATEGY_NAME_CSE_KMS, new ClientSideKMSEncryptionStrategy());
+        put(STRATEGY_NAME_CSE_CMK, new ClientSideCMKEncryptionStrategy());
+    }};
+
+    private static final AllowableValue NONE = new AllowableValue(STRATEGY_NAME_NONE, "None","No encryption.");
+    private static final AllowableValue SSE_S3 = new AllowableValue(STRATEGY_NAME_SSE_S3, "Server-side S3","Use server-side, S3-managed encryption.");
+    private static final AllowableValue SSE_KMS = new AllowableValue(STRATEGY_NAME_SSE_KMS, "Server-side KMS","Use server-side, KMS key to perform encryption.");
+    private static final AllowableValue SSE_C = new AllowableValue(STRATEGY_NAME_SSE_C, "Server-side Customer Key","Use server-side, customer-supplied key for encryption.");
+    private static final AllowableValue CSE_KMS = new AllowableValue(STRATEGY_NAME_CSE_KMS, "Client-side KMS","Use client-side, KMS key to perform encryption.");
+    private static final AllowableValue CSE_CMK = new AllowableValue(STRATEGY_NAME_CSE_CMK, "Client-side Customer Master Key","Use client-side, customer-supplied master key to perform encryption.");
+
+    public static final PropertyDescriptor ENCRYPTION_STRATEGY = new PropertyDescriptor.Builder()
+            .name("Encryption Strategy")
```

Comment:
For `.name()` we try to use no whitespace, so `encryption-strategy`.

Reply:
Updated these properties.